packages/@lwc/ssr-compiler/src/compile-js/index.ts

```diff
@@ -252,6 +252,9 @@ export default function compileJS(
     let ast = parseModule(src, {
         module: true,
         next: true,
+        loc: true,
+        source: filename,
+        ranges: true,
```

Comment:
Some possible changes to avoid bloating AST:

- It didn't look like `ranges` are used by the LSP/VSCode extension, I only included it in order to match `babel-plugin-compiler`.
- Source could be moved to the ComponentMetaState.

Reply:
Matching `babel-plugin-component` exactly here is not necessary IMO, unless it's fulfilling some function in VS Code... and I take it it's not?

Reply:
At least here it's not used: https://github.com/forcedotcom/lightning-language-server/blob/46ec16e555ff23dfb90a3daea060d70a31830720/packages/lwc-language-server/src/javascript/compiler.ts#L117

That's the only place I looked though.